### PR TITLE
Request to add GRU Dimension

### PR DESCRIPTION
**Pull request by anewman89**
I've incorporated the GRU dimension into SUMMA.  The code passed the test case checks after implementation, and again after merging the current SUMMA/develop.  Additionally, I changed the time structure to have an integer second variable isec.  Finally, to get the GRU-HRU correspondence into SUMMA, minimal netCDF input capability was added only for the GRU-HRU mapping, see addition of <read_allocate_gru_struc> in <allocspace.f90>.  To avoid changes to the fileManager file, I have a file specified that needs to be with the SUMMA exe to read the GRU-HRU mapping.  I'm sure this should be changed, so any suggestions are welcome.

### CODE REVIEW

**Comment by anewman89** [comment]
Do you know why the Travis CI build would be pulling paths from my home directory?

File: build/Makefile [hunk 1]
```diff
@@ -13,12 +13,14 @@
 
 # Define core directory below which everything resides. This is the
 # parent directory of the 'build' directory
-F_MASTER =
+F_MASTER = /home/anewman/summa
```
Comment (bartnijssen) [inline build/Makefile]: @anewman89 : This is why travis is complaining - you should remove this commit from your PR

File: build/source/engine/vegPhenlgy.f90 [hunk 2]
```diff
@@ -125,6 +125,8 @@ subroutine vegPhenlgy(&
 
  ) ! associate variables in data structure
  ! ----------------------------------------------------------------------------------------------------------------------------------
+print*, 'scalarLAI=', scalarLAI, ' scalarSAI=', scalarSAI, ' scalarExposedLAI=', scalarExposedLAI, ' scalarExposedSAI=', scalarExposedSAI, &
+        'veg type=',vegTypeIndex
```
Comment (bartnijssen) [inline build/source/engine/vegPhenlgy.f90]: Remove debug messages from PR - there are a few more like these

File: build/source/engine/time_utils.f90 [hunk 3]
```diff
@@ -30,12 +30,12 @@ module time_utils_module
  ! ******************************************************************************************
  ! public subroutine extractTime: extract year/month/day/hour/minute/second from units string
  ! ******************************************************************************************
- subroutine extractTime(refdate,iyyy,im,id,ih,imin,dsec,err,message)
+ subroutine extractTime(refdate,iyyy,im,id,ih,imin,isec,err,message)
```
Comment (bartnijssen) [inline build/source/engine/time_utils.f90]: These changes appear unrelated to the addition of the GRU dimension. Can you split them out so that the PR is self-contained?

File: build/source/engine/read_force.f90 [hunk 4]
```diff
@@ -126,7 +127,8 @@ subroutine read_force(istep,iHRU,err,message)
                   refTime%var(iLookTIME%im),              & ! input  = month
                   refTime%var(iLookTIME%id),              & ! input  = day
                   refTime%var(iLookTIME%ih),              & ! input  = hour
-                  refTime%var(iLookTIME%imin),dsec,       & ! input  = minute/second
+                  refTime%var(iLookTIME%imin),            & ! output = minute
+	          refTime%var(iLookTIME%isec),            & ! output = second
```
Comment (bartnijssen) [inline build/source/engine/read_force.f90]: Split out the time step changes from the GRU changes (or explain why they are related).

Fix indenting (below as well)

File: build/source/engine/read_attrb.f90 [hunk 5]
```diff
@@ -177,31 +192,38 @@ USE get_ixname_module,only:get_ixAttr,get_ixType  ! access function to find inde
  call get_vlines(unt,dataLines,err,cmessage)
  if(err/=0)then; message=trim(message)//trim(cmessage); return; endif
  ! get the number of HRUs
- nHRU = size(dataLines)
+ !nHRU = size(dataLines)
```
Comment (bartnijssen) [inline build/source/engine/read_attrb.f90]: Don't commit commented out code. Just remove the line.

File: build/source/engine/mDecisions.f90 [hunk 6]
```diff
@@ -200,7 +199,7 @@ subroutine mDecisions(err,message)
                   startTime%var(iLookTIME%id),                           & ! day
                   startTime%var(iLookTIME%ih),                           & ! hour
                   startTime%var(iLookTIME%imin),                         & ! minute
-                  dsec,                                                  & ! second
+                  startTime%var(iLookTIME%isec),                         & ! second
```
Comment (bartnijssen) [inline build/source/engine/mDecisions.f90]: Separate time change from the GRU PR (if possible) and motivate the change from a `real*8` to an integer second (I wonder if that can create problems for the dynamic time stepping, although that may be a different time stamp).

File: build/source/engine/coupled_em.f90 [hunk 7]
```diff
@@ -387,7 +387,7 @@ lightSnow            ! maximum interception capacity an inverse function of new
                   err,cmessage)                   ! intent(out): error control
   if(err/=0)then; err=20; message=trim(message)//trim(cmessage); return; endif
 
-
+  print*, 'dt_done=', dt_done, 'summerLAI=', mpar_data%var(48), ' scalarBelowCanopySolar=', mvar_data%var(64)%dat
```
Comment (bartnijssen) [inline build/source/engine/coupled_em.f90]: Remove debugging code

File: build/source/engine/derivforce.f90 [hunk 8]
```diff
@@ -185,6 +185,8 @@ subroutine derivforce(err,message)
  spectralIncomingDiffuse(1) = SWRadAtm*(1._dp - scalarFractionDirect)*Frad_vis              ! (diffuse vis)
  spectralIncomingDiffuse(2) = SWRadAtm*(1._dp - scalarFractionDirect)*(1._dp - Frad_vis)    ! (diffuse nir)
 
+ print*, 'derivforce: cosZenith=', cosZenith, ' scalarFractionDirect=',scalarFractionDirect, ' Frad_vis=', Frad_vis
```
Comment (bartnijssen) [inline build/source/engine/derivforce.f90]: Remove debugging code

File: build/source/engine/allocspace.f90 [hunk 9]
```diff
@@ -39,6 +40,157 @@ module allocspace_module
...
+ ! populate number of hrus within each gru and allocate space for hrus accordingly
+ do iGRU=1, nGRU
+  gru_struc(iGRU)%hruCount = buf_int_1(iGRU)
+  hruCount = gru_struc(iGRU)%hruCount  
+  allocate(gru_struc(iGRU)%hru(hruCount), stat=err)
+  if(err/=0)then; err=20; message=trim(message)//'problem allocating space for hru in gru_struc'; return; endif
+ enddo
+
+ ! get maxHRU dimension length
+! call check(nf90_inq_dimid(ncid, "maxHRU", hruDimID),message)
```
Comment (bartnijssen) [inline build/source/engine/allocspace.f90]: Don't commit commented out code (next line as well)

File: build/source/driver/multi_driver.f90 [hunk 10]
```diff
@@ -348,15 +371,21 @@ real(dp),parameter        :: safety = 0.01_dp               ! safety factor to e
...
+    mvar_gru(iGRU)%hru(iHRU)%var(iLookMVAR%scalarAquiferStorage)%dat(1) = 0._dp  ! not used
+   end do
   end do
  case default; call handle_err(20,'unable to identify decision for regional representation of groundwater')
 endselect
 
 ! initialize time step length for each HRU
-do iHRU=1,nHRU
- dt_init(iHRU) = mvar_hru(iHRU)%var(iLookMVAR%dt_init)%dat(1) ! seconds
+do iGRU=1, nGRU
```
Comment (bartnijssen) [inline build/source/driver/multi_driver.f90]: remove space after comma

**Comment by anewman89** [comment]
@bartnijssen  I've made your suggested changes, please have another look.  Thanks!

File: build/source/engine/allocspace.f90 [hunk 11]
```diff
@@ -158,34 +323,45 @@ subroutine alloc_forc(nHRU,err,message)
...
+ enddo ! end of iGRU loop
+ 
  end subroutine alloc_forc
 
 
  ! ************************************************************************************************
  ! public subroutine alloc_attr: initialize data structures for local attributes
  ! ************************************************************************************************
- subroutine alloc_attr(nHRU,err,message)
+ subroutine alloc_attr(nGRU, nHRU,err,message)
```
Comment (bartnijssen) [inline build/source/engine/allocspace.f90]: remove space after comma

**Comment by bartnijssen** [comment]
@anewman89 : much cleaner - go ahead and make the change above and I'll merge

**Comment by anewman89** [comment]
@bartnijssen Ok, made that last change.
